src/utils/ethereum.ts

```diff
@@ -69,33 +88,40 @@ async function getTokens(wallet, hermezTokens) {
 
 /**
  * Checks if an Ethereum transaction has been canceled by the user
- * @param {Object} tx - Ethereum transaction
+ * @param {TransactionResponse} tx - Ethereum transaction
  * @returns {Boolean}
  */
-function isTxCanceled(tx) {
+function isTxCanceled(tx: TransactionResponse | null): boolean {
   return tx === null;
 }
 
 /**
  * Checks if an Ethereum transaction has been mined
- * @param {Object} tx - Ethereum transaction
+ * @param {TransactionResponse} tx - Ethereum transaction
  * @returns {Boolean}
  */
-function isTxMined(tx) {
-  return tx !== null && tx.blockNumber !== null;
+function isTxMined(tx: TransactionResponse): boolean {
+  // ToDo: When can tx.blockNumber be null? I'm adding the undefined case for now.
+  return tx !== null && tx.blockNumber !== null && tx.blockNumber !== undefined;
 }
 
 /**
  * Checks if an Ethereum transaction is expected to fail. We expect a transaction to fail
  * if it exceeds a timeout (24h by default) or if the user doesn't have enough ETH in his
  * account to pay the maximum fee estimated for the tx.
- * @param {Object} tx - Ethereum transaction
+ * @param {TransactionResponse} tx - Ethereum transaction
  * @param {ISOStringDate} date - ISO string date the transaction was sent
  * @param {BigNumber} accountEthBalance - ETH balance of the account which the transaction has been sent from
  * @returns {Boolean}
  */
-function isTxExpectedToFail(tx, date, accountEthBalance) {
-  if (tx?.blockNumber === null) {
+function isTxExpectedToFail(
+  tx: TransactionResponse,
+  date: ISOStringDate,
+  accountEthBalance: BigNumber
+): boolean {
+  // ToDo: According to ethers types tx?.blockNumber can not be null
```

Comment:
Might be worth testing to confirm. I do seem to remember it returning null though. Best way is to force a tx to fail by setting a very low gas (21000)

Reply:
I feel this is difficult to test. I mean, I can confirm that it's not `null` in my tests, but how can we guarantee that it will not be `null` in the future? Unless you guys know it can be null for some strange reason, I would stick to the definition of the type from `ethers`, where the prop is optional, so its value can only be `number` or `undefined`:

![image](https://user-images.githubusercontent.com/15896177/134310497-aecca684-aaa1-4bf8-bbca-533bd754f1cf.png)

Reply:
I think @elias-garcia wrote this so he might remember better. I vaguely remember it being null before a tx has been processed

Reply:
I don't remember exactly if it's null or undefined... We would need to test it. This use case is when you put a low gas fee and the transaction hasn't been mined yet. If `ethers` says that is undefined it should be, as they use TypeScript in their codebase and if not they should have compilation errors hehe

Reply:
We've confirmed that `tx?.blockNumber` can be `null`. Comment updated.